lib/src/bitarray.dart

```diff
@@ -0,0 +1,317 @@
+  void or(BitArray ba) {
+    _checkLengthsEqual(this, ba);
+    _transformDataWords(
+        (int word, int wordIndex) => word | ba._data[wordIndex]);
+  }
+
+  /// Performs logical XOR operation to [this] with given [BitArray].
+  void xor(BitArray ba) {
+    _checkLengthsEqual(this, ba);
+    _transformDataWords(
+        (int word, int wordIndex) => word ^ ba._data[wordIndex]);
+  }
+
+  /// Creates a new instance of [BitArray] with the same values as in [this].
+  BitArray clone() {
+    var result = BitArray(_length);
+    result._setDataWords((int wordIndex) => _data[wordIndex]);
+    return result;
+  }
+
+  /// Returns a [BitArray] result of negated [ba].
+  static BitArray notGate(BitArray ba) {
+    var result = BitArray(ba.length);
+    result._setDataWords((int wordIndex) => ~ba._data[wordIndex]);
+    return result;
+  }
+
+  /// Returns a result of logical OR operation for passed [BitArray]s.
+  static BitArray orGate(BitArray ba1, BitArray ba2) {
+    _checkLengthsEqual(ba1, ba2);
+    var result = BitArray(ba1.length);
+    result._setDataWords((int wordIndex) {
+      return ba1._data[wordIndex] | ba2._data[wordIndex];
+    });
+    return result;
+  }
+
+  /// Returns a result of logical AND operation for passed [BitArray]s.
+  static BitArray andGate(BitArray ba1, BitArray ba2) {
+    _checkLengthsEqual(ba1, ba2);
+    var result = BitArray(ba1.length);
+    result._setDataWords((int wordIndex) {
+      return ba1._data[wordIndex] & ba2._data[wordIndex];
+    });
+    return result;
+  }
+
+  /// Returns a result of logical XOR operation for passed [BitArray]s.
+  static BitArray xorGate(BitArray ba1, BitArray ba2) {
+    _checkLengthsEqual(ba1, ba2);
+    var result = BitArray(ba1.length);
+    result._setDataWords((int wordIndex) {
+      return ba1._data[wordIndex] ^ ba2._data[wordIndex];
+    });
+    return result;
+  }
+
+  /// Returns [true] if bitarrays have the same length and the same values.
+  bool equal(BitArray ba) {
+    if (identical(this, ba)) return true;
+    if (_length != ba._length) return false;
+    for (var wordIndex = 0; wordIndex < _data.length; ++wordIndex) {
+      if (_data[wordIndex] != ba._data[wordIndex]) return false;
+    }
+    return true;
+  }
+
+  @override
+  bool operator [](int index) {
+    RangeError.checkValidIndex(index, this);
+    return (_data[index >> 5] & (1 << (index & 31))) != 0;
+  }
+
+  @override
+  int get length => _length;
+
+  @override
+  void operator []=(int index, bool val) {
+    RangeError.checkValidIndex(index, this);
+    if (val) {
+      _data[index >> 5] |= 1 << (index & 31);
+    } else {
+      _data[index >> 5] &= ~(1 << (index & 31));
+    }
+  }
+
+  /// Counts number of bits in unsigned [val].
+  int _countBits(int val) {
+    var bits = 0;
+    while (val != 0) {
+      bits += _numOfBitsLookup[val & 0xFF];
+      val >>= 8;
+    }
+    return bits;
+  }
+
+  /// Returns number [true] values in [this].
```

Comment:
Document getters using a noun phrase.

`[true]` and `[this]` are not valid links, they are not source names. Use `true` or `this` instead.

Maybe:
```dart
/// The number of `true` values in this bit list.
```